content-src/components/Base/Base.jsx

```diff
@@ -48,9 +48,29 @@ export class _Base extends React.PureComponent {
     this.updateTheme();
   }
 
-  componentWillUpdate({App}) {
+  hasSectionChanged(nextProps) {
+    const [nPropsSections] = nextProps.Sections;
```

Comment:
Can you do perhaps do `nextProps.Sections.find(section => section.id === "topstories")` to make this more resilient to future changes?

Additionally, it might be a good idea to call this `hasTopStoriesSectionChanged`